### PR TITLE
ABC-132: sign error page parameters

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
Sign error page parameters so client can verify they're from the server.

#### Ticket Link
https://mattermost.atlassian.net/browse/ABC-132

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -4,6 +4,7 @@
 package app
 
 import (
+	"crypto/ecdsa"
```
Comment (crspeller) [inline app/app.go]: Why ecdsa?

File: app/config.go [hunk 2]
```diff
@@ -116,8 +121,87 @@ func (a *App) InvokeConfigListeners(old, current *model.Config) {
...
+		if err != nil {
+			return err
+		}
+		system.Value = string(v)
+		if result = <-a.Srv.Store.System().Save(system); result.Err != nil {
+			key = newKey
+		}
+	}
+
+	if key == nil {
```
Comment (crspeller) [inline app/config.go]: Can we add a comment as to why this is here? You and me know, but the next person might not understand.

File: utils/api.go [hunk 3]
```diff
@@ -32,13 +35,25 @@ func OriginChecker(allowedOrigins string) func(*http.Request) bool {
 	}
 }
 
-func RenderWebError(err *model.AppError, w http.ResponseWriter, r *http.Request) {
-	status := http.StatusTemporaryRedirect
-	if err.StatusCode != http.StatusInternalServerError {
-		status = err.StatusCode
+func RenderWebAppError(w http.ResponseWriter, r *http.Request, err *model.AppError, s crypto.Signer) {
```
Comment (crspeller) [inline utils/api.go]: Return structs and accept interfaces. Just how I like it :)

File: app/app.go (hunk 1 above)
Comment (ccbrown) [reply app/app.go]: Only other practical option is RSA. And the only compelling factor in the decision was signature length. RSA signatures are much longer for comparable security. So ECDSA gives us prettier URLs. :P

**Comment by ccbrown** [comment]
@crspeller updated!

File: app/app.go (hunk 1 above)
Comment (crspeller) [reply app/app.go]: 😆 OK! Pretty URLs it is!

**Pull request merged**